src/main/java/tahub/contacts/model/course/Course.java

```diff
@@ -71,12 +71,12 @@ public boolean equals(Object other) {
             return false;
         }
 
-        return courseCode.equals(otherCourse.courseCode) && courseName.equals(otherCourse.courseName);
+        return courseCode.equals(otherCourse.courseCode);
     }
 
     @Override
     public int hashCode() {
-        return courseName.hashCode();
+        return courseCode.hashCode();
     }
```

Comment:
I understand you are changing `equals` to `hashCode` to implement checking if `Course`  with `courseCode` already exists in the SCA list. However, I think equality and hashcode should take into account both `courseCode` and `courseName`. Maybe you can try seeing if the check method exists in `Model` or you can implement it yourself by looping through the SCA list and seeing if a course with `courseCode` equal to a given `courseCode` already exists